src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -746,7 +746,7 @@ private void InterruptHandler(object sender, PinValueChangedEventArgs e)
             int mask = 1;
             int pin = 0;
 
-            while (mask < 0x10)
+            while (mask <= 0xF0)
```

Comment:
should this be `mask < 0xF0` or `mask & 0x0F != 0`?

Reply:
btw. is there a datasheet page we can take a look at here? (if so can you add a comment?)

Reply:
@krwq This is the datasheet: https://ww1.microchip.com/downloads/en/DeviceDoc/MCP23008-MCP23S08-Data-Sheet-20001919F.pdf
and on page 16 (1.6.8 - Interrupt flag INTF register) you'll see that each of the 8 bits corresponds to a pin (for the 16-pin chip there's a secondary register).
What this change does is change so the mask iterates through 0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80 where before it was just 0x01, 0x02, 0x04, 0x08 (which was the bug). You're correct that is could be `mask < 0xF0` (but not `mask & 0x0F != 0` as that wouldn't properly evaluate the correct stopping point), I just went with this, basically the `mask` will be 0x100 when the while loop should stop.

Reply:
The 0xF0 is pretty weird as the sequence never has more than a bit set.
I would have rather changed the loop in a do-while exiting at 0x80 :-)

Reply:
@krwq Or you wanted me to add the datasheet link in the commit?

Reply:
Yes, datasheet and paragraph please, for the future readers.

Reply:
@raffaeler Done!